src/wallet/rpcsigner.cpp

```diff
@@ -0,0 +1,595 @@
+
+    assert(signer != nullptr);
+    std::vector<ExternalSigner>::iterator position = std::find(pwallet->m_external_signers.begin(), pwallet->m_external_signers.end(), *signer);
+    if (position != pwallet->m_external_signers.end()) pwallet->m_external_signers.erase(position);
+
+    return NullUniValue;
+}
+
+static UniValue signerdisplayaddress(const JSONRPCRequest& request)
+{
+    if (request.fHelp || request.params.empty() || request.params.size() > 2) {
+        throw std::runtime_error(
+            "signerdisplayaddress\n"
+            "Display address on an external signer for verification.\n"
+            "\nArguments:\n"
+            "1. \"address\"       (string, required) The bitcoin address to display.\n"
+            "2. \"fingerprint\"   (string, optional) Master key fingerprint of signer\n"
+        );
+    }
+
+    std::shared_ptr<CWallet> const wallet = GetWalletForJSONRPCRequest(request);
+    CWallet* const pwallet = wallet.get();
+
+    if (!EnsureWalletIsAvailable(pwallet, request.fHelp)) {
+        return NullUniValue;
+    }
+
+    ExternalSigner *signer = GetSignerForJSONRPCRequest(request, 1, pwallet);
+
+    LOCK(pwallet->cs_wallet);
+
+    CTxDestination dest = DecodeDestination(request.params[0].get_str());
+
+    // Make sure the destination is valid
+    if (!IsValidDestination(dest)) {
+        throw JSONRPCError(RPC_INVALID_ADDRESS_OR_KEY, "Invalid address");
+    }
+
+    const CKeyMetadata* meta = nullptr;
+    CKeyID key_id = GetKeyForDestination(*pwallet, dest);
+    if (!key_id.IsNull()) {
+        auto it = pwallet->mapKeyMetadata.find(key_id);
+        if (it != pwallet->mapKeyMetadata.end()) {
+            meta = &it->second;
+        }
+    }
+    // TODO: use inferred descriptor (preferably with xpub at the account level)
+    CScript scriptPubKey = GetScriptForDestination(dest);
+    if (!meta) {
+        auto it = pwallet->m_script_metadata.find(CScriptID(scriptPubKey));
+        if (it != pwallet->m_script_metadata.end()) {
+            meta = &it->second;
+        }
+    }
+    if (!meta || meta->key_origin.IsNull()) {
+        throw JSONRPCError(RPC_WALLET_ERROR, "Missing BIP32 derivation data");
+    }
+
+    std::string path = meta->hdKeypath;
+    path.erase(0,2); // Unsafe, but to be replaced with inferred descriptor
+
+    KeyOriginInfo info;
+    if (!pwallet->GetKeyOrigin(key_id, info)) {
+        throw JSONRPCError(RPC_INTERNAL_ERROR, "Internal keypath is broken");
+    }
+
+    bool solvable = IsSolvable(*pwallet, scriptPubKey);
+    if (!solvable) {
+        throw JSONRPCError(RPC_INTERNAL_ERROR, "Address is not solvable");
+    }
+
+    std::string inferredDescriptor = InferDescriptor(scriptPubKey, *pwallet)->ToString();
+
+    // TODO: Use the inferred descriptor directly. Unfortunately it's currently
+    //       not useful (or is it??).
+
+    std::string prefix = "";
+    std::string postfix = "";
+
+    if (inferredDescriptor.find("wpkh") == 0) {
```

Comment:
`compare` is more efficient than `find` when checking prefixes, right?

What about `StartsWith(inferredDescriptor, "wpkh")` instead?

```
static inline bool StartsWith(const std::string& input, const std::string& prefix) {
    return input.size() >= prefix.size() && input.compare(0, prefix.size(), prefix) == 0;
}
```

Reply:
I plan to ditch this part of the code, so no need to improve it :-)